### DISCUSSION TITLE
Does Next Auth require to config NEXTAUTH_URL variables in env when the app start?

### DISCUSSION THREAD

**Discussion by thucxuong** [nextauthjs/next-auth#4474] [Help]
### Question 💬

I've created a basic authentication nextjs app with Next Auth with env configured properly.
The env vars work totally fine with other places except for NEXTAUTH_URL.
Everything works fine, but NextAuth throws a warning.
`[next-auth][warn][NEXTAUTH_URL]`
And I have to create an env file with NEXTAUTH_URL configured in order to make it the app run.

**Is it how the NextAuth should be configured with non-vercel environment?**

### How to reproduce ☕️

**My next.config.js file:**
```
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  env: {
    NEXTAUTH_URL: process.env.NEXTAUTH_URL,
    MONGODB_URI: process.env.MONGODB_URI,
    DATABASE_NAME: process.env.DATABASE_NAME,
    NEXTAUTH_SECRET: process.env.NEXTAUTH_SECRET,
  },
};

module.exports = nextConfig;
```

**My .env file:**
```
NEXT_URL=http://localhost:3001
NEXTAUTH_URL=http://localhost:3001
MONGODB_URI=mongodb://localhost/
DATABASE_NAME=iztalents_collector
NEXTAUTH_SECRET=sdjhfdknadsvcbuijknqefjdaysuhkjbhadfsuiaasjkbdhafjyß
```

After running `npm run build && npm start`. Everything works fine, but NextAuth throws a warning.
`[next-auth][warn][NEXTAUTH_URL]`

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by 0ubbe** [comment]
This is an expected behavior, you need to set `NEXTAUTH_URL` to whatever your production domain is so that authentication still works when deploying to your production environment 👍🏽

**Comment by thucxuong** [comment]
> This is an expected behavior, you need to set `NEXT_AUTH_URL` to whatever your production domain is so that authentication still works when deploying to your production environment 👍🏽

The thing is every other variable got replaced with final value in the built version.
Eg: `API_URL` in the code is replaced with `http://localhost:3000/api` means we cannot search `API_URL` in the implementation code. This is the way env supposed to behave. I configured NEXTAUTH_URL the same way and expect it to be replaced in the final built code. 
But the warning was thrown when I start the app, and I have to add `.env` file into production server with only NEXTAUTH_URL to make it work, which I think is not great.
